Migrated the Website to Jekyll and set-up for 0.4.0 documentation

codydaig opened the pull request:
Reference Issues #294 #443

codydaig commented:
I didn't change the content of the website, just migrated it into a modular format around jekyll and set-up the folder structure for the docs for 0.4.0. I'm completely open for feedback if a different structure or method is desired. This was just the quickest way to get it up and running to match it's current state and easily add new documentation in a modular form.

codydaig commented:
You can view how it will render on GitHub Pages here:
http://codydaig.github.io/mean/

lirantal commented on CNAME:
```diff
@@ -1 +0,0 @@
-meanjs.org
```
why is the CNAME entry removed?

codydaig replied:
I made that change after I submitted the pull request to make it render on my GitHub Pages for demonstration. I forgot that it would have added it too the pull request. My apologies. Once it's approved too be merged let me know and I'll change it back.

lirantal replied:
Cool

lirantal commented:
I guess it looks ok, but probably need another set of eyes of someone who had commited in the past to gh-pages

ilanbiala commented:
@codydaig what's the difference, functionally, between the Jekyll setup and the current gh-pages branch we have set up?

codydaig commented:
The main difference is the ability to use templates which allows areas such as the documentation to be broken up into sections. Makes the code cleaner and easier to maintain. The current website has the same header and footer elements copied into each individual html page, so if you need to make one change, you have to change every page. With jekyll, you just change it once in the template and it's changed on all pages. 
As for breaking up the code into sections, I implemented this on the documentation page. So instead of it being around 3.,000 lines of code, it's broken down into 17 pages which is easier to find, add, and modify content. Every section (designated with a link on the sidebar) is it's own .html page but is rendered to the user as one solid page just like it was before.

ilanbiala commented:
@lirantal I think it's a good idea. WDYT?

lirantal commented:
Sure, @codydaig can you please have another walk-through on the rendered page, some general sanity to see that nothing is broken and I'll be happy to merge.

ilanbiala commented:
@lirantal do we have any 0.4.0 documentation? There are a few issues relating to that.

codydaig commented:
I added the CNAME back in as well as put the relative links back to work from the root directory. (So the demo at codydaig.github.io/mean is broken but will work on gh-pages for meanjs.org) I went through and tested this thoroughly on my local machine by modifying my local hosts file. 

I also removed the sample page, post, and sass files that Jekyll comes with by default. 

This commit is good to go and ready to be merged.

lirantal commented:
thanks @codydaig , applied.
@ilanbiala I don't think there's any "proper" 0.4.0 documentation. That branch originally came from Amos a long time ago and made considerable changes there to structure, etc...
